tests/helpers/test_update_coordinator.py

```diff
@@ -70,25 +72,30 @@ def update_callback():
     assert crd.last_update_success is True
 
 
-async def test_refresh_fail(crd, caplog):
-    """Test a failing update function."""
-    crd.update_method = CoroutineMock(side_effect=update_coordinator.UpdateFailed)
+@pytest.mark.parametrize(
+    "err_msg",
+    [
+        (asyncio.TimeoutError, "Timeout fetching test data"),
+        (aiohttp.ClientError, "Error requesting test data"),
+        (update_coordinator.UpdateFailed, "Error fetching test data"),
+    ],
+)
+async def test_refresh_known_errors(err_msg, crd, caplog):
+    """Test raising known errors."""
+    crd.update_method = CoroutineMock(side_effect=err_msg[0])
 
     await crd.async_refresh()
 
     assert crd.data is None
     assert crd.last_update_success is False
-    assert "Error fetching test data" in caplog.text
+    assert err_msg[1] in caplog.text
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5120914)